lib/utils/patterns.ts

```diff
@@ -282,6 +283,69 @@ export function inferOrSelectDevice(preferredUuid: string) {
 		});
 }
 
+export function getTargetUuid(
+	sdk: BalenaSdk.BalenaSDK,
+	applicationOrDevice: string,
+): Bluebird<string> {
+	const appTest = validation.validateApplicationName(applicationOrDevice);
+	const uuidTest = validation.validateUuid(applicationOrDevice);
+
+	if (!appTest && !uuidTest) {
+		throw new Error(`Device or application not found: ${applicationOrDevice}`);
+	}
+
+	const isBalenaApplicationNotFound = (e: any) =>
+		e instanceof BalenaApplicationNotFound;
```

Comment:
This isn't necessary, you can pass the `BalenaApplicationNotFound` constructor directly the the first parameter of `.catch`